iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -226,6 +418,19 @@ void populateVMToCPatterns(MLIRContext *context,
   patterns.insert<ConstZeroOpConversion<IREE::VM::ConstI32ZeroOp>>(context);
   patterns.insert<ConstRefZeroOpConversion>(context);
 
+  // List ops
+  // TODO(simon-camp): We leak memory in the generated code, as we never release
```

Comment:
The lists themselves are reference counted so once you support !vm.ref/iree_vm_ref_t you should be fine. We can brainstorm a bit on how to emit the ref stuff unless you've got a plan already :)

Reply:
We assumed that is would be necessary to insert an `emic.call` calling `iree_vm_list_release` after the last use of the list.

Reply:
I think I've hidden my own comment with the last force push :D

> I'm not sure where I want to put the logic. Maybe a cleanup pass under .../Dialect/VM/Transforms? Suggestions welcome :)

I'm planning to walk each function and add the calls Marius mentioned, similar to [this](https://github.com/google/iree/blob/830b547fe8eedf007015b247383fa8ad55b193c9/iree/compiler/Dialect/VM/Analysis/RegisterAllocation.cpp#L265).